src/dotnet/APIView/APIViewWeb/Repositories/CosmosReviewRepository.cs

```diff
@@ -108,5 +111,126 @@ public async Task<IEnumerable<ReviewModel>> GetReviewsAsync(string serviceName,
             }
             return reviews.OrderBy(r => r.Name).ThenByDescending(r => r.LastUpdated);
         }
+
+        public async Task<IEnumerable<string>> GetReviewFirstLevelProprtiesAsync(string propertyName)
+        {
+            var query = $"SELECT DISTINCT VALUE r.{propertyName} FROM Reviews r";
+            var properties = new List<string>();
+
+            QueryDefinition queryDefinition = new QueryDefinition(query);
+            using FeedIterator<string> feedIterator = _reviewsContainer.GetItemQueryIterator<string>(queryDefinition);
+
+            while (feedIterator.HasMoreResults)
+            {
+                FeedResponse<string> response = await feedIterator.ReadNextAsync();
+                properties.AddRange(response);
+            }
+            return properties;
+        }
+
+        public async Task<(IEnumerable<ReviewModel> Reviews, int TotalCount)> GetReviewsAsync(List<string> packageNames, List<string> languages,
+            List<string> authors, List<string> tags, bool? isClosed, List<int> filterTypes, bool? isApproved, int offset, int limit)
+        {
+            (IEnumerable<ReviewModel> Reviews, int TotalCount) result = (Reviews: new List<ReviewModel>(), TotalCount: 0);
+
+            // Build up Query
+            var queryStringBuilder = new StringBuilder("SELECT * FROM Reviews r");
+            queryStringBuilder.Append(" WHERE IS_DEFINED(r.id)"); // Allows for appending the other query parts in any order

```

Comment:
any reasons why this part if not part of StringBuilder constructor?

Reply:
Review table has id column always, so I am not sure why we need this IS_DEFINED check for id.

Reply:
No particular reason just for readability.